backend/src/main/java/com/bakdata/conquery/models/auth/apitoken/ApiTokenRealm.java

```diff
@@ -0,0 +1,324 @@
+import com.bakdata.conquery.io.storage.xodus.stores.XodusStore;
+import com.bakdata.conquery.models.auth.ConqueryAuthenticationInfo;
+import com.bakdata.conquery.models.auth.ConqueryAuthenticationRealm;
+import com.bakdata.conquery.models.auth.entities.User;
+import com.bakdata.conquery.models.auth.entities.UserLike;
+import com.bakdata.conquery.models.auth.permissions.Ability;
+import com.bakdata.conquery.models.auth.util.SkippingCredentialsMatcher;
+import com.bakdata.conquery.models.config.XodusConfig;
+import com.bakdata.conquery.models.identifiable.ids.specific.UserId;
+import com.bakdata.conquery.util.io.FileUtil;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import jetbrains.exodus.ExodusException;
+import jetbrains.exodus.env.Environment;
+import jetbrains.exodus.env.EnvironmentClosedException;
+import jetbrains.exodus.env.Environments;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.shiro.authc.AuthenticationException;
+import org.apache.shiro.authc.AuthenticationToken;
+import org.apache.shiro.authc.IncorrectCredentialsException;
+import org.apache.shiro.authc.UnknownAccountException;
+import org.apache.shiro.realm.AuthenticatingRealm;
+import org.apache.shiro.util.Destroyable;
+
+
+/**
+ * This realm provides and checks long-living API tokens. The tokens support a limited scope of actions that is backed
+ * by the actual permissions of the invoking user.
+ *
+ */
+@Slf4j
+public class ApiTokenRealm extends AuthenticatingRealm implements ConqueryAuthenticationRealm, Destroyable {
+
+	private static final int ENVIRONMNENT_CLOSING_RETRYS = 2;
+	private static final int ENVIRONMNENT_CLOSING_TIMEOUT = 2; // seconds
+
+	private final Path storageDir;
+	private final XodusConfig storeConfig;
+	private final Validator validator;
+	private final ObjectMapper objectMapper;
+	private final ArrayList<XodusStore> openStoresInEnv = new ArrayList<>();
+	private final MetaStorage storage;
+	private final ApiTokenCreator apiTokenCreator = new ApiTokenCreator();
+
+	private transient Environment tokenEnvironment;
+	private transient Store<ApiTokenHash, ApiTokenData> tokenDataStore;
+	private transient Store<UUID, ApiTokenData.MetaData> tokenMetaDataStore;
+
+	public ApiTokenRealm(MetaStorage storage, Path storageDir, XodusConfig storeConfig, Validator validator, ObjectMapper objectMapper) {
+		this.storage = storage;
+		this.storageDir = storageDir;
+		this.storeConfig = storeConfig;
+		this.validator = validator;
+		this.objectMapper = objectMapper;
+		this.setCredentialsMatcher(SkippingCredentialsMatcher.INSTANCE);
+		this.setAuthenticationTokenClass(ApiToken.class);
+	}
+
+
+	@Override
+	protected void onInit() {
+		super.onInit();
+		// Open/create the database/store
+		String storeName = "api-token";
+		File tokenStore = new File(storageDir.toFile(), storeName);
+		tokenEnvironment = Environments.newInstance(tokenStore, storeConfig.createConfig());
+
+		final XodusStore data = new XodusStore(
```

Comment:
Warum baust du dir hier von Hand einen Store?

Reply:
Meines wissens nach kann Xodus auch encryption

Reply:
Wie meinst du das von hand?

Das mit der Encryption geht, dass muss man nur in der StoreConfig anpassen.